doc/documentation.md

```diff
@@ -100,6 +113,60 @@ Add a `splash_master` section to your `pubspec.yaml` file.
         branding_image: 'assets/branding_image.png'
 ```
 
+### Dark Mode Support
+
+Splash Master supports dark mode for both iOS and Android platforms. When you provide dark mode images, the system automatically picks the correct asset based on the user's theme preference.
+
+#### iOS Dark Mode
+
+For iOS, when you specify `image_dark_ios`, the generator:
+1. Creates light mode images with `luminosity: light` appearance
+2. Creates dark mode images with `luminosity: dark` appearance
+3. Updates the `Contents.json` with proper appearance entries
+
+iOS automatically switches between light and dark splash images based on the system appearance setting.
+
+#### Android Dark Mode
+
+For Android, when you specify `image_dark_android`, the generator:
+1. Places the light mode image (`splash_image.png`) in `drawable-xxhdpi/`
+2. Places the dark mode image (`splash_image_dark.png`) in the same `drawable-xxhdpi/` folder
+3. Creates `splash_screen.xml` for light mode referencing the light image
+4. Creates `splash_screen_dark.xml` for dark mode referencing the dark image
+5. Updates `values-night/styles.xml` to use `splash_screen_dark` drawable
+6. If `color_dark_android` is specified, creates `values-night/colors.xml` with the dark background color
+
+For Android 12+, when `android_12_and_above` is configured along with dark mode:
+- Light mode styles are created in `values-v31/styles.xml`
+- Dark mode styles are created in `values-night-v31/styles.xml`
+
+Android automatically switches to the dark splash screen when the system is in dark mode via the `values-night` resource qualifier.
+
+#### Example: Dark Mode Only
+
+If you only need dark mode support without other customizations:
+
+```yaml
+splash_master:
+  image: 'assets/splash_light.png'
+  image_dark_android: 'assets/splash_dark.png'
+  image_dark_ios: 'assets/splash_dark.png'
+```
+
+#### Example: Dark Mode with Background Colors
+
+For complete dark mode support with different background colors:
+
+```yaml
+splash_master:
+  image: 'assets/splash_light.png'
+  color: '#FFFFFF'
+  image_dark_android: 'assets/splash_dark.png'
+  color_dark_android: '#000000'
```

Comment:
android color needs to be used in splash_screen.xml to have any effect. Currently the color splashBgColor is not being used so it will not have any effect on splash screen background color